### ISSUE TITLE
Formats containing trailing whitespace returned differently between GetFormats and GetSearchResults

### ISSUE THREAD

**Issue by nigel-bmlt** [bmlt-enabled/bmlt-root-server#451]
A administrator had created a BMLT meeting format with trailing whitespace.

GetSearchResults (jsonp) returns the format string for this meeting as:
`    "formats": "C,LGBTQI  ,NC,GB",
`

However the GetFormats call does not return the trailing whitespace.
  ```
{
    "key_string": "LGBTQI",
    "name_string": "LGBTQI",
    "description_string": "This meeting is focused on the needs of Lesbian, Gay, Bisexual, Transgender, Queer and Intersex members of NA.",
    "lang": "en",
    "id": "10",
    "world_id": "GL",
    "root_server_uri": "https://na.org.au/main_server",
    "format_type_enum": "FC3"
  }

```

This causes lookups based on a simple 'split format string by comma' to fail.

Whilst this can be worked around on the client side, ideally we'd sanitise on input to prevent this from occuring (if not a historical issue).

**Comment by jbraswell** [comment]
We should strip the white space in the API before returning a response, and we should also strip before saving to the database.

**Comment by alanb2718** [comment]
We should also think about how to handle existing formats with leading or trailing whitespace. I like the idea of a migration function that removes extra whitespace from all existing entries. If we did that, we would just need to sanitize the input; no need to do anything with the output. However, if we leave existing whitespace in the database, it doesn't seem that useful to start sanitizing new entries only.

**Comment by jbraswell** [comment]
Agreed. Lets do that.

**Comment by cosmastech** [comment]
I realize it's a heavier lift, but could we not build a one-to-many relationship between a meeting and its meeting types? I think it's a much cleaner approach to do a query like:

```sql
SELECT * FROM `meetings` WHERE `meeting_id` IN (SELECT `meeting_id` FROM `meeting_meeting_types` WHERE `meeting_type` = 'tc')
```

than to use REGEX within a query. I'm still trying to fully grasp the codebase, so I don't know how big of a change this will be.

**Comment by jbraswell** [comment]
I agree it would be better to use a proper db schema, but given the nature of this legacy codebase, small schema changes have far reaching implications. Let's have a discussion about the right approach to modernize the codebase. We will discuss when we talk later today.

**Comment by jbraswell** [comment]
We should retest this in 3.0.0. It might not be fixed, but could behave differently since we rewrote these endpoints.
